docs/index.rst

```diff
@@ -6,8 +6,98 @@
+
+Both of them need to be *importable by the zoo* and the metadata and checks filenames have to match (the zoo imports the metadata files based on the python check filename).
+
+Metadata
+--------
+
+Metadata is a yaml file with a list of errors your checks yield. There are two compulsory parameters in the header:
+
+- :code:`namespace`
+- :code:`category`
+
+Then you list the errors with parameters:
+
+- :code:`id` - a unique identifier (*required*)
+- :code:`title` - a human-readable title (*required*)
+- :code:`description` - additional information about the error (*optional*)
+
+  - :code:`{details}` - object you can fill with information when yielding the error. Then you pass the :code:`{details}` object to the description where it is rendered (*optional*)
+
+**Metadata Structure:**
+
+.. literalinclude:: ..\zoo\auditing\standards\dummy_standards\metadata\check_constantly.yml
+    :language: yaml
+
+**Real life example:**
+
+.. code-block:: yaml
+
+    namespace: coafile
+    category: Configuration of coala
+    ---
+    - id: create_coafile
+      severity: warning
+      effort: high
+      title: Use coala to run static analysis checks on your repo
+      description: |
+        Add a `.coafile` so that people can see what rules and checks they should follow
+        when contributing to your repository,
+        and run [coala](https://coala.io) to find and fix these issues automatically.
+
+Checks
+------
+
+The zoo starts all functions in this file that have the prefix :code:`check_` (if you need any helper function just name them anything else).
+
+All :code:`check_` functions need to have an input parameter :code:`context`.
+
+:code:`context` is an object of CheckContext class (`definition <https://github.com/kiwicom/the-zoo/blob/master/zoo/auditing/runner.py>`_), the object consists of all the required information about the code that is checked.
+
+The most important parameter of the class is :code:`path`, it holds the information about where the source code is exposed.
+
+All :code:`check_` functions need to return an iterator. The recommended way is to yield generator as soon as you discover errors.
+
+The structure of the generator is: :code:`context.Result("namespace:id", True/False, {details})` where
+
+- :code:`namespace` is the namespace from your metadata file
+- :code:`id` is the id of the raised error from your metadata file
+- :code:`True/False` determines if the error was found (true if it was, false if it was not).
+- :code:`details` is an object you can fill with information when yielding the error. Then you pass the :code:`{details}` object to the description where it is rendered (*optional*).
+
+**Check function structure:**
+
+.. literalinclude:: ..\zoo\auditing\standards\dummy_standards\checks\check_constantly.py
+    :language: python
+
+**Real life example:**
+
+.. code-block:: python
+
+    def check_dockerignore(context):
+    if not (context.path / "Dockerfile").is_file():
```

Comment:
The code is not correctly indented. The `def check_dockerignore(context):` should be one level higher than the rest. It was probably copied wrong from GitHub.